### PR TITLE
[MOB-6891] OOTB view customization

### PR DESCRIPTION
**Pull request by evantk91**
## 🔹 Jira Ticket(s) if any

* [MOB-6889](https://iterable.atlassian.net/browse/MOB-6889)
* [MOB-6890](https://iterable.atlassian.net/browse/MOB-6890)
* [MOB-6891](https://iterable.atlassian.net/browse/MOB-6891)
* [MOB-6892](https://iterable.atlassian.net/browse/MOB-6892)

## ✏️ Description

> This PR implements OOTB view functionality. Introduces `IterableEmbeddedViewType` class, `IterableEmbeddedViewConfig` class to add predefined customization options, and the `IterableEmbeddedView` which implements the OOTB view as three different types: banner, card, and notification.

[MOB-6891]: https://iterable.atlassian.net/browse/MOB-6891?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[MOB-6889]: https://iterable.atlassian.net/browse/MOB-6889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOB-6890]: https://iterable.atlassian.net/browse/MOB-6890?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MOB-6892]: https://iterable.atlassian.net/browse/MOB-6892?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/embedded/IterableEmbeddedView.kt [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        return view
+    }
+
+    private fun setDefaultAction(view: View, message: IterableEmbeddedMessage) {
+        if(message.elements?.defaultAction != null) {
+            val clickedUrl = message.elements?.defaultAction?.data.takeIf { it?.isNotEmpty() == true } ?: message.elements?.defaultAction?.type
+
+            view.setOnClickListener {
+                IterableApi.getInstance().embeddedManager.handleEmbeddedClick(message, null, clickedUrl)
+                IterableApi.getInstance().trackEmbeddedClick(message, null, clickedUrl)
```
Comment (Ayyanchira) [inline iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/embedded/IterableEmbeddedView.kt:154]: Should we call trackEmbeddedClick for those EM which do not have action handlers?

Comment (Ayyanchira) [reply iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/embedded/IterableEmbeddedView.kt:154]: Check on iOS side as well

File: iterableapi-ui/src/main/res/layout-v21/banner_view.xml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            android:layout_width="wrap_content"
+            android:orientation="vertical">
+
+            <TextView
+                android:id="@+id/embedded_message_title"
+                android:layout_width="240dp"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="16dp"
+                android:layout_marginTop="16dp"
+                android:text="Try Premium"
```
Comment (Ayyanchira) [inline iterableapi-ui/src/main/res/layout-v21/banner_view.xml:29]: We should leave no trace of Fiterable app in SDK as we move it to be used as a general OOTB component

File: iterableapi-ui/src/main/res/layout-v21/banner_view.xml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                android:textSize="16sp"
+                android:textStyle="bold" />
+
+            <TextView
+                android:id="@+id/embedded_message_body"
+                android:layout_width="179dp"
+                android:layout_height="32dp"
+                android:layout_marginStart="16dp"
+                android:layout_marginTop="8dp"
+                android:text="Enjoy 14 days of Premium Subscription free" />
```
Comment (Ayyanchira) [inline iterableapi-ui/src/main/res/layout-v21/banner_view.xml:39]: Lets remove message here

File: iterableapi-ui/src/main/res/layout-v21/banner_view.xml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        </LinearLayout>
+
+        <ImageView
+            android:id="@+id/embedded_message_image"
+            android:layout_width="70dp"
+            android:layout_height="70dp"
+            android:layout_marginTop="16dp"
+            android:layout_marginBottom="16dp"
+            android:layout_marginEnd="16dp"
+            android:contentDescription="Flex kittens"
```
Comment (Ayyanchira) [inline iterableapi-ui/src/main/res/layout-v21/banner_view.xml:50]: Same as above

File: iterableapi-ui/src/main/res/layout-v21/banner_view.xml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        app:alignItems="flex_start">
+
+        <Button
+            android:id="@+id/embedded_message_first_button"
+            android:layout_width="wrap_content"
+            android:layout_height="36dp"
+            android:layout_marginStart="16dp"
+            android:background="@drawable/banner_button_background"
+            android:stateListAnimator="@null"
+            android:text="Try subscription"
```
Comment (Ayyanchira) [inline iterableapi-ui/src/main/res/layout-v21/banner_view.xml:70]: Here

File: iterableapi-ui/src/main/res/layout-v21/banner_view.xml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            app:layout_flexBasisPercent="50%" />
+
+        <Button
+            android:id="@+id/embedded_message_second_button"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:backgroundTint="@android:color/transparent"
+            android:stateListAnimator="@null"
+            android:textColor="@color/white"
+            android:text="Try subscription"
```
Comment (Ayyanchira) [inline iterableapi-ui/src/main/res/layout-v21/banner_view.xml:83]: Here as well

File: iterableapi-ui/src/main/res/layout-v21/card_view.xml
Comment (Ayyanchira) [inline iterableapi-ui/src/main/res/layout-v21/card_view.xml:1]: Same comments as for other xml. We can get rid of default Fiterable messages in our OOTB components

File: iterableapi-ui/src/main/res/layout-v21/notification_view.xml
Comment (Ayyanchira) [inline iterableapi-ui/src/main/res/layout-v21/notification_view.xml:1]: Same as above

File: iterableapi-ui/src/main/res/values/strings.xml
Comment (Ayyanchira) [inline iterableapi-ui/src/main/res/values/strings.xml:1]: Should be part of Fiterable

**Review by Ayyanchira — COMMENTED** [review]
Added some changes

File: iterableapi-ui/src/main/res/drawable/fiterable_entertainment_popper.png
Comment (evantk91) [inline iterableapi-ui/src/main/res/drawable/fiterable_entertainment_popper.png:1]: deleting

File: iterableapi-ui/src/main/res/drawable/fiterable_boxing.png
Comment (evantk91) [inline iterableapi-ui/src/main/res/drawable/fiterable_boxing.png:1]: deleting

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/embedded/IterableEmbeddedView.kt (hunk 1 above)
Comment (evantk91) [reply iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/embedded/IterableEmbeddedView.kt:154]: default action is null when the open action is set to "none". So this code takes care of preventing click tracking calls. We can review the iOS side.

File: iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/embedded/IterableEmbeddedView.kt [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+import com.iterable.iterableapi.IterableApi
+import com.iterable.iterableapi.IterableEmbeddedMessage
+import com.iterable.iterableapi.IterableLogger
+import com.iterable.iterableapi.ui.R
+
+class IterableEmbeddedView(
+    private var viewType: IterableEmbeddedViewType,
+    private var message: IterableEmbeddedMessage,
+    private var config: IterableEmbeddedViewConfig?
+): Fragment() {
```
[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi-ui/src/main/java/com/iterable/iterableapi/ui/embedded/IterableEmbeddedView.kt:24]: Tackle later for now fragments are used.

File: iterableapi-ui/build.gradle [hunk 8]
```diff
@@ -30,12 +30,15 @@ dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     implementation 'androidx.appcompat:appcompat:1.0.0'
     implementation 'androidx.recyclerview:recyclerview:1.0.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation 'com.google.android.flexbox:flexbox:3.0.0'
+    implementation 'androidx.cardview:cardview:1.0.0'
+    implementation "com.github.bumptech.glide:glide:4.8.0"
```
[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi-ui/build.gradle:36]: Tackle later

**Review by evantk91 — COMMENTED** [review]
Requesting review.

**Review by Ayyanchira — APPROVED** [review]
LGTM

**Pull request merged**